.github/workflows/main.yml

```diff
@@ -8,22 +8,38 @@ env:
 jobs:
   build:
     name: Build
-    runs-on: ubuntu-18.04
+    runs-on:  ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
+        os: [ubuntu-20.04]
         orocos_build_type: [Debug, Release]
         compiler: [gcc, clang]
+        python_version: ['3.8']
+        include:
+          - os: ubuntu-18.04
+            orocos_build_type: Release
+            compiler: gcc
+            python_version: '2'
+          - os: ubuntu-20.04
+            orocos_build_type: Release
+            compiler: gcc
+            python_version: '3.9'
     env:
       CC: ${{ matrix.compiler }}
       OROCOS_BUILD_TYPE: ${{ matrix.orocos_build_type }}
+      ROS_PYTHON_VERSION: ${{ matrix.python_version }}
     steps:
       - uses: actions/checkout@v2
         with:
           submodules: recursive
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
       - name: Install
         run: |
-          sudo apt-get install libeigen3-dev libcppunit-dev python-psutil python3-psutil python-future python3-future
+          sudo apt-get install libeigen3-dev libcppunit-dev
+          pip install psutil future
```

Comment:
It is required because of the setup-python environment. When you use another version of python system packages are not compatible. These are all 3.8 for 20 04. So when using a different version, it conflicts.